tutorial/markdown/java/spring-boot.md

```diff
@@ -73,152 +122,261 @@ Once the site is up and running you can launch your browser and go to the Swagge
+6. The `AirlineController` constructs an HTTP response with a status code of 200 OK and includes the updated airline data in the response body as a JSON object.
+7. The response is sent back to the client with the updated airline data if found, or a 404 Not Found response if the airline with the specified ID does not exist.
+
+### DELETE Route Workflow
+
+`@DeleteMapping("/{id}")`
+
+The DELETE route is triggered when a client sends an HTTP DELETE request to `/api/v1/airline/{id}`, where `{id}` is the unique identifier of the airline to be deleted.
+
+1. The `AirlineController` receives the request and extracts the `id` from the URL path.
+2. The `AirlineController` calls the `deleteAirline` method of the `AirlineService`, passing the `id` of the airline to be deleted.
+3. The `AirlineService` is responsible for deleting the airline from the database using the `AirlineRepository`.
+4. If the airline is deleted successfully, the `AirlineService` performs the deletion operation without returning any response data.
+5. The `AirlineController` constructs an HTTP response with a status code of 204 No Content, indicating that the request was successful, but there is no content to return in the response body.
+6. The response is sent back to the client to indicate the successful deletion of the airline.
+
+These workflows illustrate how each HTTP method interacts with the `AirlineService` and the underlying database through the `AirlineRepository` to perform various operations on airline data.
+
+## Custom SQL++ Queries
+
+1. Get all airlines by country
 
 ```java
-profileCol.remove(id.toString());;
-```
 
-> *from Delete method of controllers/ProfileController.java*
+    @Override
+    public List<Airline> findByCountry(String country, int limit, int offset) {
+      String statement = "SELECT airline.id, airline.type, airline.name, airline.iata, airline.icao, airline.callsign, airline.country FROM `"
+                + dbProperties.getBucketName() + "`.`inventory`.`airline` WHERE country = '" + country + "' LIMIT "
+                + limit + " OFFSET " + offset;
+        return cluster
+                .query(statement, QueryOptions.queryOptions().scanConsistency(QueryScanConsistency.REQUEST_PLUS)
+                        .parameters(JsonObject.create().put("country", country)))
+                .rowsAs(Airline.class);
+
+    }
+```
 
-## GET Profiles by Searching
+<!-- Explaination -->
 
-[N1QL](https://docs.couchbase.com/dotnet-sdk/current/howtos/n1ql-queries-with-sdk.html) is a powerful query language based on SQL, but designed for structured and flexible JSON documents. We will use a N1QL query to search for profiles with Skip, Limit, and Search options.
+In the above example, we are using the `QueryOptions` class to set the `scanConsistency` to `REQUEST_PLUS` to ensure that the query returns the latest data. We are also using the `JsonObject` class to set the `country` parameter in the query.
```

Comment:
Links to docs for these methods/options would be helpful

Reply:
Added links to docs